examples/pools/main.rs

```diff
@@ -61,40 +57,34 @@ fn main() {
         events::handle_instantiate_event,
     );
 
-    // Define a generic handler for transactions,
-    // which the processor will call for each transaction.
-    fn transaction_handler(
+    #[transaction_handler]
+    async fn transaction_handler(
         context: TransactionHandlerContext<AppState, TxHandle>,
     ) -> Result<(), TransactionHandlerError> {
-        // Do something like start a database transaction
-        let mut transaction_context =
-            context.app_state.async_runtime.block_on(async {
-                // start a database transaction
-                let tx = context.app_state.pool.begin().await.unwrap();
-                TxHandle { transaction: tx }
-            });
+        let tx = context.app_state.pool.begin().await.unwrap();
+        let mut transaction_context = TxHandle { transaction: tx };
 
         // Handle the events in the transaction
-        context.transaction.handle_events(
-            context.app_state,
-            context.handler_registry,
-            &mut transaction_context,
-        )?;
+        context
+            .incoming_transaction
+            .handle_events(
+                context.app_state,
+                context.handler_registry,
+                &mut transaction_context,
+            )
+            .await?;
 
         // Commit the database transaction
-        context.app_state.async_runtime.block_on(async {
-            transaction_context.transaction.commit().await.unwrap();
-        });
+        transaction_context.transaction.commit().await.unwrap();
         Ok(())
     }
 
     if arg == "file" {
```

Comment:
In general, better to use a match here for the different cases and have each case in its own function